doc: add module README for Firefox and derivatives (#897)
danth/stylix: modules/firefox/README.md

> # Firefox and derivatives

Reviewer: This might be more explicit:

```suggestion
# Firefox and Firfox derivatives
```

> This module supports [Firefox](https://www.mozilla.org/firefox/), and its
> derivatives [Floorp](https://floorp.app/) and
> [LibreWolf](https://librewolf.net/), using the same underlying implementation.
> Each browser has its own set of options.

Reviewer: What about the following:

```suggestion
This module supports [Firefox](https://www.mozilla.org/firefox), and its
[LibreWolf](https://librewolf.net) and [Floorp](https://floorp.app) derivatives.

Although all derivatives share the same implementation, they do not share option
values.
```
